packages/nvidia_nat_adk/src/nat/plugins/adk/llm.py

```
@@ -194,27 +182,53 @@ async def dynamo_adk(config: DynamoModelConfig, _builder: Builder):
+
+        prediction_lookup: PredictionTrieLookup | None = None
+        if config.nvext_prediction_trie_path:
+            try:
+                trie_path = Path(config.nvext_prediction_trie_path)
+                trie = load_prediction_trie(trie_path)
+                prediction_lookup = PredictionTrieLookup(trie)
+                logger.info("Loaded prediction trie from %s", config.nvext_prediction_trie_path)
+            except FileNotFoundError:
+                logger.warning("Prediction trie file not found: %s", config.nvext_prediction_trie_path)
+            except Exception as e:
+                logger.exception("Failed to load prediction trie: %s", e)
```

Ruff (0.15.2)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)